src/main/java/com/redhat/rhjmc/containerjfr/net/WebServer.java

```diff
@@ -234,14 +233,14 @@ private Response serveClientIndex() {
 
         @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
         private Response serveClientAsset(String assetName) {
-            Path assetPath = Paths.get("/", "web-client", assetName);
-            if (!assetPath.toFile().isFile()) {
+            URL url = WebServer.class.getResource(assetName);
+            if (url == null) {
                 return newFixedLengthResponse(Status.NOT_FOUND, NanoHTTPD.MIME_PLAINTEXT,
                         String.format("%s not found", assetName));
             }
             try {
-                String mime = NanoHTTPD.getMimeTypeForFile(assetPath.toUri().toString());
-                InputStream assetStream = Files.newInputStream(assetPath);
+                String mime = NanoHTTPD.getMimeTypeForFile(url.toString());
```

Comment:
In retrospect, passing the full file path to `getMimeTypeForFile` was probably not any better than simply passing in `assetName`. Please try that out, and if it still works as expected, let's go with that simple way instead.

Reply:
Examing the source of `getMimeTypeForFile`, it determines to extension name by looking for the last occurrence of `.` in the input. Therefore, regardless of the input being a full URL, a pathname, a base name, or even just an extension name begins with `.`, it returns the same mine-type.

I agree with skipping obtaining `url` and using `assetName` directly.